File: subworkflows/1_prep_samples/code/mouse.R
get_contigsfile <- function(dir, names_file){
  p <- dir
  p <- gsub("\\/$", "", p)
  f <- list.files(path=p, pattern="*.fastq.gz")
  f <- f[!grepl("Mock", f)] #let's ignore the mock community data

  r1 <- f[grep("_R1_", f)]
  r1.group <- sub("_S.*", "", r1)

  r2 <- f[grep("_R2_", f)]
  r2.group <- sub("_S.*", "", r2)

  stopifnot(r1.group == r2.group)

  files.file <- paste0(gsub(".*\\/(.*)", "\\1", p), ".files")
  p.files.file <- paste0(p, "/", files.file)
  output_table <- cbind(r1.group, r1, r2)
  write.table(file=names_file, output_table, sep="\t", quote=F, row.names=F, col.names=F)
}

get_contigsfile("data/mouse/raw/", snakemake@output[["file"]])

File: subworkflows/1_prep_samples/code/mouse.py
import collections
import os

if 'snakemake' in locals() or 'snakemake' in globals():
    print("Using filepaths from snakemake")
    input_files = snakemake.input['files']
    output_file = snakemake.output['file']
else:
    print("Using default filepaths")
    input_files = os.listdir("data/mouse/raw")
    output_file = 'data/mouse/mouse.files'

print("Mapping names to files")
names = collections.defaultdict(list)
for file in sorted(input_files):
    name = file.split('/')[-1].split('_')[0]
    if "mock" not in name.lower():
        names[name].append(file)

print(f"Writing {output_file}")
with open(output_file, 'w') as file:
    for name, files in names.items():
        file_list = '\t'.join(files)
        file.write(f"{name}\t{file_list}\n")